app/tests/test_queries.py

```diff
@@ -0,0 +1,210 @@
+    """
+    Given a user accessing the browse view query
+    When they make a POST request with a transferring body filter
+    Then it should return records matched to transferring body filter.
+    """
+    create_two_test_records()
+    bodies = Body.query.all()
+
+    if len(bodies) > 0:
+        transferring_body = bodies[0].BodyId
+        search_results = browse_data(transferring_body_id=transferring_body)
+        assert len(search_results) == 1
+
+
+def test_browse_data_with_series_filter(client: FlaskClient):
+    """
+    Given a user accessing the browse view query
+    When they make a POST request with a series filter
+    Then it should return records matched to series filter.
+    """
+    create_two_test_records()
+    series = Series.query.all()
+
+    if len(series) > 0:
+        series_id = series[0].SeriesId
+        search_results = browse_data(series_id=series_id)
+        assert len(search_results) == 1
+
+
+def test_browse_data_with_consignment_reference_filter(client: FlaskClient):
+    """
+    Given a user accessing the browse view query
+    When they make a POST request with a consignment reference filter
+    Then it should return records matched to consignment reference filter.
+    """
+    create_two_test_records()
+    consignments = Consignment.query.all()
+
+    if len(consignments) > 0:
+        consignment_id = consignments[0].ConsignmentId
+        search_results = browse_data(consignment_id=consignment_id)
+        assert len(search_results) == 1
+
+
+@patch("app.main.db.queries.db")
+def test_browse_data_exception_raised(db, capsys):
+    """
+    Given a browse data query
+    When a call made to browse data , when database execution failed with error
+    Then list should be empty and should raise an exception
+    """
+
+    def mock_execute(_):
+        raise exc.SQLAlchemyError("foo bar")
+
+    db.session.execute.side_effect = mock_execute
+    results = browse_data()
+    assert results == []
+    assert (
+        "Failed to return results from database with error : foo bar"
+        in capsys.readouterr().out
+    )
+
+
+def test_get_file_meta_data_no_results(client: FlaskClient):
+    """
+    Given a user with a search query
+    When they make a request on the browse view series page, and no results are found
+    Then they should see no records found.
+    """
+    file_id = "junk"
+    search_results = get_file_meta_data(file_id)
+
+    assert len(search_results) == 0
+
+
+def test_get_file_meta_data_with_results(client: FlaskClient):
+    """
+    Given a user with a search query which should return n results
+    When they make a request on the browse series page
+    Then a table is populated with the n results with metadata fields.
+    """
+    create_two_test_records()
+    files = File.query.all()
+
+    if len(files) > 0:
+        file_id = files[0].FileId
+        search_results = get_file_meta_data(file_id=file_id)
+        assert len(search_results) == 3
```

Comment:
Since we only need one File to test this query, I think we can simplify the setup by just creating one File with some related file metadata objects rather than creating 2 records and then grabbing one of them from the db.

Again, once we start using FactoryBoy we can make test setup simpler. Could be worth incorporating it here but for now, to get this merged in quicker can just copy one part of that create two test records function.

Either way, I think it would be worth making the docstring more explciit so its easier to understand whats happening E.g.:

```
    """
    Given a File object with 3 related metadata objects
    When get_file_meta_data is called with the file's id
    Then 3 FileMetadata results are returned
    """
```